src/main/java/uk/gov/hmcts/cmc/claimstore/controllers/support/SupportController.java

```diff
@@ -163,6 +167,76 @@ public void uploadDocumentToDocumentManagement(
                 .orElseThrow(() -> new NotFoundException("Unable to upload the document. Please try again later")));
     }
 
+    @PostMapping("/claim/{referenceNumber}/reset-operation")
+    @ApiOperation("Redo any failed operation. Use the claim submission indicators to indicate the operation to redo.")
+    public void resetOperation(
+        @PathVariable("referenceNumber") String referenceNumber,
+        @RequestBody ClaimSubmissionOperationIndicators claimSubmissionOperationIndicators,
+        @RequestHeader(value = HttpHeaders.AUTHORIZATION) String authorisation
+    ) {
+        if (StringUtils.isBlank(authorisation)) {
+            throw new BadRequestException(AUTHORISATION_IS_REQUIRED);
+        }
+        Claim claim = claimService.getClaimByReferenceAnonymous(referenceNumber)
+            .orElseThrow(() -> new NotFoundException(String.format(CLAIM_DOES_NOT_EXIST, referenceNumber)));
+        validateClaimSubmissionOpsInd(claim.getClaimSubmissionOperationIndicators(),
+            claimSubmissionOperationIndicators);
+        claim = claimService.updateClaimSubmissionOperationIndicators(
+            authorisation,
+            claim,
+            claimSubmissionOperationIndicators
+        );
+        triggerAsyncOperation(authorisation, claim);
+    }
+
+    private void validateClaimSubmissionOpsInd(
+        ClaimSubmissionOperationIndicators existingClaimSubmissionOperationIndicators,
+        ClaimSubmissionOperationIndicators newClaimSubmissionOperationIndicators
+    ) {
+        List<String> invalidIndicators = new ArrayList<>();
+
+        if (existingClaimSubmissionOperationIndicators.getClaimIssueReceiptUpload().equals(NO)
```

Comment:
There was a comment which seems lost now. 
It was to move all this validation out of controller, can have a ConstraintValidator which we can apply on the request payload. And it can have it's own unit tests to cover all scenarios.

Reply:
@sabahirfan yes Indeed and I had replied to the comment as well. I don't know how it got lost. In my reply I had asked , if you can point me to a custom validator which does similar validations. In the application, I see only constraint validators and they work slightly differently. For this particular validation, we need to fetch the claim first whereas I don't see that for other constraint validators. Can you please point me to a sample one where we do this kind of operation. Also the unit tests are already there . Please have a look at SuppotControllerTest.java

Reply:
You can find plenty of examples under this package `uk.gov.hmcts.cmc.domain.constraints`. 
Unit test for the validation in controller test is wrong place.

Reply:
I have already seen some of the validators in this package.. please can you point me to one where we are fetching the claim and validating against the data in the claim.

Reply:
Payload validation should be moved in a constraint, if there is more than that create a Rule class and move that logic there. You can find examples of Rules under `uk.gov.hmcts.cmc.claimstore.rules`

Reply:
I do not agree with this comment. I see validation being done in other methods in this controller as well.